Base/QTCore/qSlicerCoreApplication.h

```diff
@@ -573,6 +573,24 @@ class Q_SLICER_BASE_QTCORE_EXPORT qSlicerCoreApplication : public QApplication
   /// Leaves other paths unchanged.
   Q_INVOKABLE QStringList toSlicerHomeRelativePaths(const QStringList& path) const;
 
+  /// This method can be called by Slicer core or extensions to record a software usage event.
+  /// The application does just emits a 'usageEventLogged' signal and it is up to modules to
+  /// use this information, for example to compute software usage statistics.
+  /// \param component is 'core' for events logged by Slicer core modules, and the extension name for
+  /// events logged by modules in that extension.
+  /// \param event is the name of the event.
+  /// The event name must be shorter than 50 characters to avoid potential performance degradation.
+  /// The event name must not contain any information about the user or any of the processed data to
+  /// alleviate any privacy concerns when handling software usage data.
+  /// It is recommended to use all lowercase and use '/' as separator if events can be organized
+  /// in a hierarchical structure (for example, 'planning/model created', 'planning/model exported', 'segmentation/ct/total', 'segmentation/mr/knee').
+  ///
+  /// For example, an extension can report usage data like this:
+  /// - C++: <code>qSlicerCoreApplication::application()->logUsageEvent("SlicerRT", "dicom/export/sro")</code>
+  /// - Python: <code>slicer.app.logUsageEvent("TotalSegmentator", "segmentation/total")</code>
```

Comment:
Instead of using `/` as delimiter, I suggest to tweak event name like this:
* use `.` like it is done in https://doc.qt.io/qt-6/qloggingcategory.html
* only accept letter, digit and `_`

Reply:
Should we differentiate between a function called as an effect of using the UI ? and as an effect of using an API call ?

Following today's discussion, developer may add a dedicated gui event.